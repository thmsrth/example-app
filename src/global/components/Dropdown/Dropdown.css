:root {
  /** Colors **/
  --dropdown-icon-color: var(--color-dark-grey);
  --dropdown-content-background-color: var(--color-white);
  --dropdown-button-padding: var(--element-padding);
  --dropdown-button-background-color: var(--color-brand-secondary);
}

.Dropdown {
  position: relative;
}

.Dropdown .Dropdown__content {
  position: absolute;
  overflow: hidden !important;
  max-height: 0;
  width: 100%;
}

.Dropdown--active .Dropdown__content {
  max-height: 1000px;
  transition: max-height .5s ease-in-out;
}

.Dropdown__container {
  position: relative;
}

.Dropdown__icon {
  font-size: 1.2em;
  color: var(--dropdown-icon-color);
}

.Dropdown__content {
  z-index: 1;
  background-color: var(--dropdown-content-background-color);
}

.Dropdown__button {
  background-color: var(--dropdown-button-padding);
}
