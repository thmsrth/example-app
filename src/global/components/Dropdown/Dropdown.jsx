import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

import { bemCls } from '../../utils/ClassNameHelpers';

import './Dropdown.css';

const Dropdown = (props) => {
  const {
    isActive, className, text, onClick, children,
  } = props;

  return (
    <div
      className={`Dropdown ${className} ${
        bemCls(`Dropdown ${className}`, '--active', isActive)}`}
    >
      <div className={bemCls(`Dropdown ${className}`, '__container')}>
        <div
          className={bemCls(`Dropdown ${className}`, '__button')}
          onClick={onClick}
          onKeyPress={onClick}
          role="button"
          tabIndex="0"
        >
          <div className={bemCls(`Dropdown ${className}`, '__text')}>{text}</div>
          <div className={bemCls(`Dropdown ${className}`, '__icon')}>
            <Icon name={`${isActive ? 'caret-up' : 'caret-down'}`} />
          </div>
        </div>
        <div
          className={bemCls(`Dropdown ${className}`, '__content')}
          onClick={onClick}
          onKeyPress={onClick}
          role="button"
          tabIndex="0"
        >
          {children}
        </div>
      </div>
    </div>
  );
};

Dropdown.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
  isActive: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.node,
};

Dropdown.defaultProps = {
  className: '',
  text: '',
  isActive: false,
  onClick: () => {},
  children: <h3>Empty dropdown content</h3>,
};

export default Dropdown;
