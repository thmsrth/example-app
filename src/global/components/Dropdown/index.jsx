/**
 * Dropdown Component for React
 *
 * Description:
 * This component is a simple dropdown trigger which displays its children in a container.
 *
 * Additional Features:
 * - closeOnClickOutside (Boolean), true per default
 *
 * Dependencies:
 * - react-onclickoutside
 *
 * Example usage :
 * <Dropdown
 *   text={currentItem.title}
 *   className="ItemsListDropdown"
 *   closeOnClickOutside: false
 *   children=<ListOfItems items={items} />
 * />
 *
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import onClickOutside from 'react-onclickoutside';

import Dropdown from './Dropdown';

class DropdownWrapper extends Component {
  constructor() {
    super();
    this.state = {
      isActive: false,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(prevState => ({ isActive: !prevState.isActive }));
  }

  handleClickOutside() {
    if (this.props.closeOnClickOutside && this.state.isActive) {
      this.setState({ isActive: false });
    }
  }

  render() {
    const { isActive } = this.state;
    return (
      <Dropdown
        {...this.props}
        isActive={isActive}
        onClick={this.handleClick}
      />
    );
  }
}

DropdownWrapper.propTypes = {
  ...Dropdown.propTypes,
  closeOnClickOutside: PropTypes.bool,
};

DropdownWrapper.defaultProps = {
  ...Dropdown.defaultProps,
  closeOnClickOutside: true,
};

export default onClickOutside(DropdownWrapper);
