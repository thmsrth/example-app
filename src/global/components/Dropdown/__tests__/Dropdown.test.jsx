import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Dropdown from '../Dropdown';

const props = {
  className: 'MyDropdown',
  text: 'MyText',
  isActive: false,
  onClick: () => {},
  children: <h3>Empty dropdown content</h3>,
};

describe('Dropdown', () => {
  it('should render the component correctly while inactive', () => {
    const tree = shallow(<Dropdown {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly while active', () => {
    const tree = shallow(<Dropdown {...props} isActive />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
