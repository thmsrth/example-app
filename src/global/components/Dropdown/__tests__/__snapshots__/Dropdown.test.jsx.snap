// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render the component correctly while active 1`] = `
<div
  className="Dropdown MyDropdown Dropdown--active MyDropdown--active"
>
  <div
    className="Dropdown__container MyDropdown__container"
  >
    <div
      className="Dropdown__button MyDropdown__button"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="Dropdown__text MyDropdown__text"
      >
        MyText
      </div>
      <div
        className="Dropdown__icon MyDropdown__icon"
      >
        <Icon
          className=""
          name="caret-up"
        />
      </div>
    </div>
    <div
      className="Dropdown__content MyDropdown__content"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <h3>
        Empty dropdown content
      </h3>
    </div>
  </div>
</div>
`;

exports[`Dropdown should render the component correctly while inactive 1`] = `
<div
  className="Dropdown MyDropdown "
>
  <div
    className="Dropdown__container MyDropdown__container"
  >
    <div
      className="Dropdown__button MyDropdown__button"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="Dropdown__text MyDropdown__text"
      >
        MyText
      </div>
      <div
        className="Dropdown__icon MyDropdown__icon"
      >
        <Icon
          className=""
          name="caret-down"
        />
      </div>
    </div>
    <div
      className="Dropdown__content MyDropdown__content"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <h3>
        Empty dropdown content
      </h3>
    </div>
  </div>
</div>
`;
